src/MonoTorrent/MonoTorrent.Client/PeerConnections/SocketConnection.cs

```diff
@@ -43,19 +43,19 @@ public class SocketConnection : IConnection
 
         public bool CanReconnect => !IsIncoming;
 
-        public bool Connected => Socket.Connected;
+        public bool Connected => SK.Connected;
 
         EndPoint IConnection.EndPoint => EndPoint;
 
-        public IPEndPoint EndPoint { get; }
+        public IPEndPoint EndPoint { get; private set; }
 
         public bool IsIncoming { get; }
 
-        SocketAsyncEventArgs ReceiveArgs { get; }
+        protected SocketAsyncEventArgs ReceiveArgs { get; private set; }
 
-        SocketAsyncEventArgs SendArgs { get; }
+        protected SocketAsyncEventArgs SendArgs { get; private set; }
 
-        Socket Socket { get; }
+        protected Socket SK { get; private set; }
```

Comment:
Do we need to add the `protected` modifier to these? I'd rather leave them private unless there's a really important reason why they have to be public. People are free to implement `IConnection` themselves if they need to do weird things with the underlying sockets. Also, if the SocketAsyncEventArgs are made public then it becomes harder to rewrite the internals of this class to use a different underlying async mechanism in future!